src/qt/transactiontablemodel.cpp

```diff
@@ -120,6 +120,20 @@ class TransactionTablePriv
 
         switch(status)
         {
+        case CT_UPDATED_FULL:
+        case CT_DELETED:
+            if(!inModel)
+            {
+                qWarning() << "TransactionTablePriv::updateWallet: Warning: Got CT_DELETED, but transaction is not in model";
+                break;
+            }
+            // Removed -- remove entire transaction from table
+            parent->beginRemoveRows(QModelIndex(), lowerIndex, upperIndex-1);
+            cachedWallet.erase(lower, upper);
+            parent->endRemoveRows();
+            if (status == CT_DELETED)
+                break;
+            inModel = false;
```

Comment:
Please do not implicitly fall through. I recommend inverting the if statement if you meant to break here after inModel = false.

Reply:
Fall through is intentional here to prevent code duplication, since `CT_UPDATED_FULL` is equivalent to first `CT_DELETED` and then `CT_NEW`.

Reply:
Ok, then please add a comment explaining the fallthrough is intentional.

Reply:
Okay, I'll do that